File: atcoder/abc135B_0_or_1_swap.ml
(* Vicfred
 * https://atcoder.jp/contests/abc135/tasks/abc135_b
 * greedy
 * *)
let n = Scanf.scanf "%d\n" ( + ) 0
let ps = Array.init n @@ fun _ -> Scanf.scanf " %d" ( + ) 0
let main =
    print_endline @@ 
    if ( >= ) 2 @@
    Array.fold_left ( + ) 0 @@
    Array.init n @@ fun i -> if (i + 1) != ps.(i) then 1 else 0
    then "YES"
    else "NO"

File: atcoder/abc135B_0_or_1_swap.nim
# uninhm
# https://atcoder.jp/contests/abc135/tasks/abc135_b
# implementation

import strutils, sequtils, algorithm

let
  n = stdin.readLine.parseInt
  p = stdin.readLine.split.map parseInt

var p2 = p
p2.sort

var diff = 0
for i in 0 ..< n:
  if p[i] != p2[i]:
    inc diff
  if diff > 2:
    break

if diff > 2 or diff == 1:
  echo "NO"
else:
  echo "YES"
